src/main/scala/org/apache/spark/sql/pulsar/PulsarHelper.scala

```diff
@@ -58,8 +56,7 @@ private[pulsar] case class PulsarHelper(
   private var topicPartitions: Seq[String] = _
 
   override def close(): Unit = {
-    CachedPulsarClient.clear()
-    CachedConsumer.clear()
+    // do nothing
```

Comment:
will `CachedPulsarClient` and `CachedConsumer` be cleared in another place?

Reply:
Good point. 

During my test, if they are cleared in the `close()` call , the whole job will be terminated before consuming any data. Notice the 4th last line.

```
23/03/30 14:34:49 INFO CachedConsumer: Cleaning up Consumer Cache.
23/03/30 14:34:49 INFO PulsarProvider: Schema of Pulsar source: StructType(StructField(value,BinaryType,false), StructField(__key,BinaryType,true), StructField(__topic,StringType,true), StructField(__messageId,BinaryType,true), StructField(__publishTime,TimestampType,true), StructField(__eventTime,TimestampType,true), StructField(__messageProperties,MapType(StringType,StringType,true),true))
23/03/30 14:34:50 INFO StateStoreCoordinatorRef: Registered StateStoreCoordinator endpoint
23/03/30 14:34:50 INFO ResolveWriteToStream: Checkpoint root /tmp/2cf8c1f8-e420-494c-b0f0-cbdb2f431343 resolved to file:/tmp/2cf8c1f8-e420-494c-b0f0-cbdb2f431343.
23/03/30 14:34:50 WARN ResolveWriteToStream: spark.sql.adaptive.enabled is not supported in streaming DataFrames/Datasets and will be disabled.
23/03/30 14:34:50 INFO CheckpointFileManager: Writing atomically to file:/tmp/2cf8c1f8-e420-494c-b0f0-cbdb2f431343/metadata using temp file file:/tmp/2cf8c1f8-e420-494c-b0f0-cbdb2f431343/.metadata.12de7bde-7504-4d09-b7cd-dcae51b6778b.tmp
23/03/30 14:34:50 INFO CheckpointFileManager: Renamed temp file file:/tmp/2cf8c1f8-e420-494c-b0f0-cbdb2f431343/.metadata.12de7bde-7504-4d09-b7cd-dcae51b6778b.tmp to file:/tmp/2cf8c1f8-e420-494c-b0f0-cbdb2f431343/metadata
23/03/30 14:34:50 INFO MicroBatchExecution: Starting [id = b03cd4e6-8990-4328-85de-cf51aff0dade, runId = a7b5361c-c785-4b0b-a678-1057efffda67]. Use file:/tmp/2cf8c1f8-e420-494c-b0f0-cbdb2f431343 to store the query checkpoint.
23/03/30 14:34:50 INFO SparkContext: Invoking stop() from shutdown hook
23/03/30 14:34:50 INFO PulsarConfigUpdater: pulsar.client: Set 'service.url' to 'pulsar://localhost:6650', earlier value: 'pulsar://localhost:6650'
23/03/30 14:34:50 INFO PulsarConfigUpdater: pulsar.reader: Set 'subscriptionName' to 'nengnengneng', earlier value: 'nengnengneng'
23/03/30 14:34:50 INFO PulsarConfigUpdater: pulsar.admin: Set 'service.url' to 'pulsar://localhost:6650', earlier value: 'pulsar://localhost:6650'
```